tailscale_compliance/tailscale.sp

```diff
@@ -0,0 +1,241 @@
+locals {
+  tailscale_common_tags = merge(local.tailscale_compliance__common_tags, {
+    tailscale = "true"
+  })
+}
+
+variable "api_key_id" {
+  type        = string
+  description = "API Key ID for the tailnet key."
+  default     = "kFXfcN2CNTRL"
+}
+
+benchmark "tailscale" {
+  title       = "Tailscale Best Practices"
+  description = "Tailscale has many security features you can use to increase your network security. This page provides best practices for using these features to harden your Tailscale deployment."
+
+  children = [
+    control.tailscale_tailnet_acl_groups_used,
+    control.tailscale_tailnet_acl_tags_used,
+    control.tailscale_acl_ssh_admin_roles_assigned,
+    control.tailscale_acl_ssh_check_mode_enabled,
+    control.tailscale_device_authorization_enabled,
+    control.tailscale_device_key_expire,
+    control.tailscale_device_network_boundary_protected,
+    control.tailscale_tailnet_key_unused,
+    control.tailscale_device_upgrade_clients_in_timely_manner
+  ]
+
+  tags = merge(local.tailscale_common_tags, {
+    type = "Benchmark"
+  })
+}
+
+control "tailscale_tailnet_acl_groups_used" {
+  title       = "Use groups in ACLs"
+  description = "Use tags to manage devices. Tags allows to define access to devices based on purpose, rather than based on owner. ."
+  sql =<<-EOT
+    select
+      tailnet_name as resource,
+      case
+        when acl_groups is not null then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when acl_groups is not null then 'Tailnet use acl groups.'
+        else 'Tailnet does not use acl groups.'
+      end as reason,
+      dns_preferences
```

Comment:
What's in `dns_preferences`? This column is JSON type, so it doesn't seem like a great additional dimension.

We typically include identifier information here instead to help give context, especially when using multiple connections.